dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/PrivateDataManager.kt

```diff
@@ -301,4 +268,64 @@ class PrivateDataManager(
     fun getDocumentFromInMemoryStore(hash: String): ByteArray? {
         return documentInMemoryStorage[hash]
     }
+
+    /**
+     * Retrieves a private sme data object from the private storage
+     * @param dataId the dataId of the dataset to be retrieved
+     * @param correlationId the correlationId of the request
+     */
+    fun getPrivateDataSet(dataId: String, correlationId: String): StorableDataSet {
+        return dataManagerUtils.getDataSet(
+            dataId, DataType.of(SmeData::class.java), correlationId,
+            ::getDataFromCacheOrStorageService,
+        )
+    }
+    private fun getDataFromCacheOrStorageService(dataId: String, correlationId: String): String {
+        return jsonDataInMemoryStorage[dataId] ?: dataManagerUtils
+            .getDataFromStorageService(
+                dataId, correlationId,
+                ::getPrivateData,
+            )
+    }
+    private fun getPrivateData(dataId: String, correlationId: String): String {
+        return storageClientUtils.getDataByIdFromStorageClient(dataId, correlationId)
+    }
+
+    /**
+     * This method retrieves a document from the storage
+     * @param dataId the dataId connected to document to be retrieved
+     * @param hash the hash of the requested document
+     * @param correlationId the correlationId of the request
+     */
+    @Suppress("TooGenericExceptionCaught")
+    fun retrievePrivateDocumentById(dataId: String, hash: String, correlationId: String): DocumentStream {
+        // TODO Wenn man nach einem document sucht, was es nicht gibt, entsteht hier aktuell ein internal server error
+        // TODO Grund: Die nächste Zeile kriegt eine out of bounds exception.  Das müssen wir iwie anders handlen,
+        // TODO Vorschlag: Passende exception schmeißen (ResourceNotFoundException z.B.)
+        val eurodatId: String
+        try {
+            eurodatId = dataIdToHashMappingRepository.findByDataIdAndHash(dataId, hash)[0].eurodatId
+        } catch (e: IndexOutOfBoundsException) {
+            logger.info("Dataset with id $dataId could not be found. Correlation ID: $correlationId")
+            throw ResourceNotFoundApiException(
+                "Document not found",
+                "No document for dataId $dataId with the hash: $dataId could be found in the data store.",
```

Comment:
Achtung: Hinter "hash" steht nochmal die dataId.

Aber abgesehen davon:
Streng genommen sucht er ja nicht im data store, sondern in den Dataland-eigenen mappings.
Würde dann vllt eher sowas loggen wie

"Dataland cannot match the dataId $dataId and hash $dataId to a eurodatId to retrieve the document from EuroDaT"

Was denkst du?

Reply:
Gefällt mir die Änderung, ist eingebaut.

Reply:
"Document not found" steht da noch und stimmt nicht ganz.
Eher "No matching eurodatId found"

Reply:
Geändert